go-runtime/compile/build.go

```diff
@@ -87,14 +93,15 @@ func (b ExternalModuleContext) NonMainModules() []*schema.Module {
 	return modules
 }
 
-const buildDirName = "_ftl"
+const buildDirName = ".ftl"
+const DefaultGoModVersion = "1.21"
```

Comment:
There should never be a need for something like this, we should always pull the Go version from the module's `go.mod` file.

Reply:
This was something I wanted to chat about. With this new shared setup, I'm not sure which version to use here. Given that we might be generating shared modules for multiple modules. Or in the initial case we're generating shared modules for just `builtins` so I don't have a go version to get from a module. In the other cases we're very likely generating a single shared `.ftl/go/modules` for a multiple modules, so I wasn't sure which go version to use there either.

The big one is the builtins though. That's what led to me adding this here, but it felt dirty for sure. I also added this central one so we could share it with the `cmd_new.go` go code that was using a hardcoded version in it's template.

Reply:
Ah good question. In that case we should use the highest version.